systemvm/patches/debian/config/opt/cloud/bin/cs/CsHelper.py

```diff
@@ -179,12 +179,12 @@ def get_hostname():
         return line.strip()
 
 
-def execute(command):
+def execute(command, wait=True):
     """ Execute command """
     logging.debug("Executing: %s" % command)
     p = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
-    result = p.communicate()[0]
-    return result.splitlines()
+    if wait:
+        return p.communicate()[0].splitlines()
```

Comment:
@romain-dartigues can you make it return a `return []` after line 187. This would handle the case when wait if `False`, and still return empty array which may be consumed/necessary the the caller of the method. Please consider any other change necessary to handle the case that John has mentioned.